cli/internal/cmd/applyinit.go

```diff
@@ -0,0 +1,238 @@
+		var nonRetriable *nonRetriableError
+		if errors.As(err, &nonRetriable) {
+			cmd.PrintErrln("Cluster initialization failed. This error is not recoverable.")
+			cmd.PrintErrln("Terminate your cluster and try again.")
+			if nonRetriable.logCollectionErr != nil {
+				cmd.PrintErrf("Failed to collect logs from bootstrapper: %s\n", nonRetriable.logCollectionErr)
+			} else {
+				cmd.PrintErrf("Fetched bootstrapper logs are stored in %q\n", a.flags.pathPrefixer.PrefixPrintablePath(constants.ErrorLog))
+			}
+		}
+		return nil, err
+	}
+	a.log.Debugf("Initialization request successful")
+
+	a.log.Debugf("Buffering init success message")
+	bufferedOutput := &bytes.Buffer{}
+	if err := a.writeOutput(stateFile, resp, a.flags.mergeConfigs, bufferedOutput, measurementSalt); err != nil {
+		return nil, err
+	}
+
+	return bufferedOutput, nil
+}
+
+// initCall performs the gRPC call to the bootstrapper to initialize the cluster.
+func (a *applyCmd) initCall(ctx context.Context, dialer grpcDialer, ip string, req *initproto.InitRequest) (*initproto.InitSuccessResponse, error) {
+	doer := &initDoer{
+		dialer:   dialer,
+		endpoint: net.JoinHostPort(ip, strconv.Itoa(constants.BootstrapperPort)),
+		req:      req,
+		log:      a.log,
+		spinner:  a.spinner,
+		fh:       file.NewHandler(afero.NewOsFs()),
+	}
+
+	// Create a wrapper function that allows logging any returned error from the retrier before checking if it's the expected retriable one.
+	serviceIsUnavailable := func(err error) bool {
+		isServiceUnavailable := grpcRetry.ServiceIsUnavailable(err)
+		a.log.Debugf("Encountered error (retriable: %t): %s", isServiceUnavailable, err)
+		return isServiceUnavailable
+	}
+
+	a.log.Debugf("Making initialization call, doer is %+v", doer)
+	retrier := retry.NewIntervalRetrier(doer, 30*time.Second, serviceIsUnavailable)
+	if err := retrier.Do(ctx); err != nil {
+		return nil, err
+	}
+	return doer.resp, nil
+}
+
+// generateMasterSecret reads a base64 encoded master secret from file or generates a new 32 byte secret.
+func (a *applyCmd) generateMasterSecret(outWriter io.Writer) (uri.MasterSecret, error) {
+	// No file given, generate a new secret, and save it to disk
+	key, err := crypto.GenerateRandomBytes(crypto.MasterSecretLengthDefault)
+	if err != nil {
+		return uri.MasterSecret{}, err
+	}
+	salt, err := crypto.GenerateRandomBytes(crypto.RNGLengthDefault)
+	if err != nil {
+		return uri.MasterSecret{}, err
+	}
+	secret := uri.MasterSecret{
+		Key:  key,
+		Salt: salt,
+	}
+	if err := a.fileHandler.WriteJSON(constants.MasterSecretFilename, secret, file.OptNone); err != nil {
+		return uri.MasterSecret{}, err
+	}
+	fmt.Fprintf(outWriter, "Your Constellation master secret was successfully written to %q\n", a.flags.pathPrefixer.PrefixPrintablePath(constants.MasterSecretFilename))
+	return secret, nil
+}
+
+// writeOutput writes the output of a cluster initialization to the
+// state- / kubeconfig-file and saves it to disk.
+func (a *applyCmd) writeOutput(
```

Comment:
nit: to indicate that it's only relevannt for init?
```suggestion
func (a *applyCmd) writeInitOutput(
```